modules/apim/product/README.md

```diff
@@ -0,0 +1,105 @@
+| `terms`                | `string` | Yes      | Configures the legal terms to accept prior to subscribing to the product. |
+| `groups`               | `array`  | No       | Configures a list of groups that are added to the product.                |
+
+## Outputs
+
+| Name              | Type   | Description                                                                  |
+| :---------------- | :----: | :--------------------------------------------------------------------------- |
+| name              | string | The name of the API Management Product                                       |
+| id                | string | A unique identifier for the API Management Product.                          |
+| resourceGroupName | string | The name of the Resource Group where the API Management Product is deployed. |
+| subscriptionId    | string | The guid for the subscription where the API Management Product is deployed.  |
+
+## Examples
+
+### Example 1
+
+A basic example of a API Management product which requires subscriptions and is not published by default.
+
+```bicep
+@description('A reference to an existing APIM service.')
+resource service 'Microsoft.ApiManagement/service@2022-08-01' existing = {
+  name: 'my-apim-service-001'
+}
+
+@description('A basic product which requires subscriptions and is not published by default.')
+module reporting 'br/public:apim/product:1.0.0' = {
+  name: 'reporting-apis'
+  params: {
+    displayName: 'Reporting'
+    description: 'A group of APIs for managing reports.'
+    serviceId: service.id
+    terms: 'By using this product you agree to the terms of service.'
+  }
+}
+```
+
+### Example 2
+
+An example of a API Management product which requires subscriptions and is automatically published.
+
+```bicep
+@description('A reference to an existing APIM service.')
+resource service 'Microsoft.ApiManagement/service@2022-08-01' existing = {
+  name: 'my-apim-service-001'
+}
+
+@description('A product which requires subscriptions and is automatically published.')
+module reporting 'br/public:apim/product:1.0.0' = {
+  name: 'reporting-apis'
+  params: {
+    displayName: 'Reporting'
+    description: 'A group of APIs for managing reports.'
+    serviceId: service.id
+    terms: 'By using this product you agree to the terms of service.'
+    published: true
+    approvalRequired: true
+    subscriptionRequired: true
+  }
+}
+```
+
+### Example 3
+
+An example of a API Management product which automatically grants the `internal-developers` group access to view and subscribe to the product.
+
+```bicep
+@description('A reference to an existing APIM service.')
+resource service 'Microsoft.ApiManagement/service@2022-08-01' existing = {
+  name: 'my-apim-service-001'
+}
+
+@description('A basic product which automatically grants the `internal-developers` group access to view and subscribe to the product.')
+module reporting 'br/public:apim/product:1.0.0' = {
```

Comment:
It's a bit confusing, but the module version will always start with `MAJOR.MINOR.1`.
```suggestion
module reporting 'br/public:apim/product:1.0.1' = {
```